crates/types/src/simple_vote.rs

```diff
@@ -0,0 +1,225 @@
+        self.data.commit()
+    }
+}
+
+impl<TYPES: NodeType, DATA: Voteable + 'static, MEMBERSHIP: Membership<TYPES>>
+    SimpleVote<TYPES, DATA, MEMBERSHIP>
+{
+    /// Creates and signs a simple vote
+    pub fn create_signed_vote(
+        data: DATA,
+        view: TYPES::Time,
+        pub_key: &TYPES::SignatureKey,
+        private_key: &<TYPES::SignatureKey as SignatureKey>::PrivateKey,
+    ) -> Self {
+        let signature = TYPES::SignatureKey::sign(private_key, data.commit().as_ref());
+        Self {
+            signature: (pub_key.to_bytes(), signature),
+            data,
+            current_view: view,
+            _pd: PhantomData,
+        }
+    }
+}
+
+impl<LEAF: Committable> Committable for YesData<LEAF> {
+    fn commit(&self) -> Commitment<Self> {
+        commit::RawCommitmentBuilder::new("Yes Vote")
+            .var_size_bytes(self.leaf_commit.as_ref())
+            .finalize()
+    }
+}
+
+impl<BLOCK: Committable> Committable for DAData<BLOCK> {
+    fn commit(&self) -> Commitment<Self> {
+        commit::RawCommitmentBuilder::new("DA Vote")
+            .var_size_bytes(self.block_commit.as_ref())
+            .finalize()
+    }
+}
+impl<BLOCK: Committable> Committable for VIDData<BLOCK> {
+    fn commit(&self) -> Commitment<Self> {
+        commit::RawCommitmentBuilder::new("DA Vote")
+            .var_size_bytes(self.block_commit.as_ref())
+            .finalize()
+    }
+}
+
+/// This implements commit for all the types which contain a view and relay public key.
+fn view_and_relay_commit<TYPES: NodeType, T: Committable>(
+    view: TYPES::Time,
+    relay: &EncodedPublicKey,
+    tag: &str,
+) -> Commitment<T> {
+    let builder = commit::RawCommitmentBuilder::new(tag);
+    builder
+        .var_size_field("Relay public key", &relay.0)
+        .u64(*view)
+        .finalize()
+}
+
+impl<TYPES: NodeType> Committable for ViewSyncPreCommitData<TYPES> {
+    fn commit(&self) -> Commitment<Self> {
+        view_and_relay_commit::<TYPES, Self>(self.round, &self.relay, "View Sync Precommit")
+    }
+}
+
+impl<TYPES: NodeType> Committable for ViewSyncFinalizeData<TYPES> {
+    fn commit(&self) -> Commitment<Self> {
+        view_and_relay_commit::<TYPES, Self>(self.round, &self.relay, "View Sync Finalize")
+    }
+}
+impl<TYPES: NodeType> Committable for ViewSyncCommitData<TYPES> {
+    fn commit(&self) -> Commitment<Self> {
+        view_and_relay_commit::<TYPES, Self>(self.round, &self.relay, "View Sync Commit")
+    }
+}
+
+// impl votable for all the data types in this file sealed marker should ensure nothing is accidently
+// implemented for structs that aren't "voteable"
+impl<V: sealed::Sealed + Committable + Clone + Serialize + Debug + PartialEq + Hash + Eq> Voteable
+    for V
+{
+}
+
+// Type aliases for simple use of all the main votes.  We should never see `SimpleVote` outside this file
+/// Yes vote Alias
+pub type YesVote<TYPES, LEAF, M> = SimpleVote<TYPES, YesData<LEAF>, M>;
```

Comment:
Similar to @elliedavidson's comment about `YesData`, can we just call this `QuorumVote`?